### ISSUE TITLE
with marks selected, toggling a mark and then reselecting marks unmarks all marks

### ISSUE THREAD

**Issue by britt-orville** [BestImageViewer/geeqie#1226]
**Setup (please complete the following information):**
 - Distribution:debian
 - Distribution release: bookworm
 - Geeqie version [geeqie --version]: geeqie         1:2.0.1-8+b1 amd64

**Describe the bug**
when i mark several images, let's say with "1" and then do "ctrl+1" to select all those marked 1 it correctly selects the marked images.
then, if i, with the images selected, press "1" to unmark one of the selected images it unmarks the image, but the selection of images remains the same. so then if i subsequently press ctrl+1 again to select only marked images, all images with mark 1 are unmarked.

i am guessing this is not by design?

**To reproduce**

**Expected behavior**
i would expect the marked images to be selected, not unmarked.

**Screenshots**

**Image sources**
If applicable, upload image files causing the bug. (Use gzip)

**Error logs or seg. fault files**
If applicable, upload text files from the bug. (Use gzip)

**Additional context**

**Comment by xsdg** [comment]
(Adding step numbers to the repro case)
> 1) when i mark several images, let's say with "1"
> 2) and then do "ctrl+1" to select all those marked 1 it correctly selects the marked images.
> 3) then, if i, with the images selected, press "1" to unmark one of the selected images it unmarks the image, but the selection of images remains the same
> 4) so then if i subsequently press ctrl+1 again to select only marked images, all images with mark 1 are unmarked.

I'm not in a situation to attempt to repro this right now, but I believe Geeqie's baseline behavior is that if multiple images are selected, pressing any number to toggle a mark will toggle that mark for _all_ of the selected images.  So I would expect all of the selected images to remove the `1` mark at step 3.

My sense is that the main bug here is that _any_ images remain showing after step 3.

**Comment by britt-orville** [comment]
ahh ok!
so the correct workflow would be to shift-ctrl+A to deselect all, then toggle the mark for the current image, then ctrl+1 to select marked again!

**Comment by xsdg** [comment]
Right.  Or if you use the mouse, clicking the mark checkbox in the file list should cause that image to automatically disappear while leaving the other images selected.

**Comment by xsdg** [comment]
Oh, actually, I just realized I'm confusing two different features; my apologies.  I'm confusing marks _filtering_ (which only shows images in the file list that have a given mark) and mark _selection_ (which selects images with a specific mark, but doesn't otherwise modify the list of images shown).

So my last comment was incorrect, and @britt-orville 's last comment is the correct way to accomplish this.  My apologies for any confusion.

**Comment by britt-orville** [comment]
great! thank you very much for your assistance!

**Comment by xsdg** [comment]
Okay, now that I'm taking a look at this, it seems like Geeqie is behaving correctly, and the confusion was around the effect of pressing a number (to toggle the respective mark) with multiple images selected.  So my sense is that this is behaving as intended.  Does that seem accurate, @britt-orville ?
